File: 2016/gara4/ciclismo/att/ciclismo.pas
function pedala(N, M: longint; var H, da, a: array of longint): longint;
begin
    (* Mettete qui il codice della soluzione *)
    pedala := 42;
end;


const
    MAXN = 10000;
    MAXM = 100000;
var
    N, M, i : longint;
    H       : array[0..MAXN-1] of longint;
    da      : array[0..MAXM-1] of longint;
    a       : array[0..MAXM-1] of longint;
    fr, fw  : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, M);
    for i:=0 to N-1 do
        read(fr, H[i]);
    readln(fr);
    for i:=0 to M-1 do
        readln(fr, da[i], a[i]);

    writeln(fw, pedala(N, M, H, da, a));
    close(fr);
    close(fw);
end.

File: ed2/gara4/ciclismo/att/ciclismo.c
#include <stdio.h>
#include <assert.h>

#define MAXN 10000
#define MAXM 100000

int pedala(int N, int M, int H[], int da[], int a[]) {
    // Mettete qui il codice della soluzione
    return 42;
}


int H[MAXN], da[MAXM], a[MAXM];

int main() {
    FILE *fr, *fw;
    int N, M, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d %d", &N, &M));
    for(i=0; i<N; i++)
        assert(1 == fscanf(fr, "%d", &H[i]));
    for(i=0; i<M; i++)
        assert(2 == fscanf(fr, "%d %d", &da[i], &a[i]));

    fprintf(fw, "%d\n", pedala(N, M, H, da, a));
    fclose(fr);
    fclose(fw);
    return 0;
}
